scene/2d/node_2d.cpp

```diff
@@ -395,6 +401,16 @@ Point2 Node2D::to_global(Point2 p_local) const {
 	return get_global_transform().xform(p_local);
 }
 
+String Node2D::get_configuration_warning() const {
+
+	Vector2 abs_scale = get_scale().abs();
+	if (abs_scale.x < CMP_EPSILON || abs_scale.y < CMP_EPSILON) {
+		return TTR("A Node2D node cannot have a scale with a zero component.");
```

Comment:
These comparisons would also return true if the scale is negative. I don't know if negative scales are allowed before reaching this code, but this is something to keep in mind. If negative scales should also error out here, perhaps change the text to "A Node2D node must have a positive non-zero scale.". If you only wish to check for zero scales, having the [is_zero_approx](https://github.com/godotengine/godot/pull/18992) method would help.